sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -1449,6 +1449,12 @@ func (host *nodeLanguageHost) RunPlugin(
 		return err
 	}
 
+	if len(env) == 0 {
+		env = os.Environ()
+	}
+	env = append(env, "PULUMI_ROOT_DIRECTORY="+req.Info.RootDirectory)
+	env = append(env, "PULUMI_PROGRAM_DIRECTORY="+req.Info.ProgramDirectory)
```

Comment:
This bit is unfortunate. We have this in the handshake, but in the provider we need to read the schema before, e.g. https://github.com/tgummerer/pulumi-tls-self-signed-cert/blob/tg/test/lib/provider.ts#L70.  Not sure if we can do better somehow?

Reply:
Hrm it was because the old version of the implementation read a `schema.json` from disk, so it needed to know where it could find that. But it looks like we no longer need that, so I think we can get rid of this.

Reply:
My new version still reads package.json. But isn't it true that an application can always find its own files (with __dirname__ or whatnot)?

Reply:
No, the working directory for the application is set to the programs root directory.  That said, looking at the code, reading that is done in the provider, at which point the handshake already happened, so the engine already gave that directory to the program.  So I think we're good here.